src/base/bittorrent/peerinfo.cpp

```diff
@@ -208,11 +208,10 @@ qlonglong PeerInfo::totalDownload() const
 QBitArray PeerInfo::pieces() const
 {
     QBitArray result(m_nativeInfo.pieces.size());
-
-    int i = 0;
-    for (const bool bit : m_nativeInfo.pieces)
-        result.setBit(i++, bit);
-
+    for (int i = 0; i < result.size(); ++i) {
+        if (m_nativeInfo.pieces[i])
```

Comment:
Unfortunately it won't compile for me due to this line.

Reply:
what was the error message?

Reply:
The error is due the "strongly <del>fucked</del> typed" indexes used by libtorrent-1.2.

Reply:
Strange... I'm not affected by it. Anyway, do the following work for you?
```c++
QBitArray PeerInfo::pieces() const
{
#if (LIBTORRENT_VERSION_NUM < 10200)
    using PieceIndex = int;
#else
    using PieceIndex = lt::piece_index_t;
#endif

    QBitArray result(m_nativeInfo.pieces.size());
    for (int i = 0; i < result.size(); ++i) {
        if (m_nativeInfo.pieces[PieceIndex {i}])
            result.setBit(i, true);
    }
    return result;
}
```

Reply:
Cannot test it now. But seems it should work (since this is the same way as we use in other similar cases).

Reply:
>Strange... I'm not affected by it. 

What libtorrent branch do you use?
I suppose Travis doesn't fail due the fact we still use deprecated libtorrent API there.

Reply:
>What libtorrent branch do you use?

RC_1_2.
Hotfix is at PR #12121.

Reply:
>RC_1_2

Have you deprecated API disabled?

Reply:
>Have you deprecated API disabled?

I don't define TORRENT_NO_DEPRECATE.

Reply:
>I don't define TORRENT_NO_DEPRECATE.

Have you any reason to do it?
Vexed... So you can't produce code that's guaranteed to be compatible with libtorrent-1.2 (native API), and Travis won't help you detect it.
@sledgehammer999, please update CI environments ASAP.

Reply:
>Have you any reason to do it?

I was using a public libtorrent package that is intended for users and I need to turn that switch on every time I rebuild it and probably overlooked. I agree at least CI should have it enabled.